velox/exec/MergeJoin.cpp

```diff
@@ -565,7 +577,34 @@ RowVectorPtr MergeJoin::getOutput() {
         // No rows survived the filter. Get more rows.
         continue;
       } else {
-        return output;
+        if (isAntiJoin(joinType_)) {
+          auto numRows = output->size();
+          const auto& filterRows = leftJoinTracker_->matchingRows(numRows);
+          auto numPassed = 0;
+
+          BufferPtr indices = allocateIndices(numRows, pool());
+          auto rawIndices = indices->asMutable<vector_size_t>();
+          for (auto i = 0; i < numRows; i++) {
+            if (!filterRows.isValid(i)) {
+              rawIndices[numPassed++] = i;
+            }
+          }
+
+          if (numPassed == 0) {
+            // No rows passed.
+            return nullptr;
+          }
+
+          if (numPassed == numRows) {
+            // All rows passed.
+            return output;
+          }
+
+          // Some, but not all rows passed.
+          return wrap(numPassed, indices, output);
+        } else {
+          return output;
```

Comment:
I wonder if it's better to make let this branch return first to improve the readability
```
If (!isAntiJoin(joinType_)) {
  return output;
}
auto numRows = output->size();
...
```